core/app/models/spree/shipping_rate.rb

```diff
@@ -17,9 +17,11 @@ class ShippingRate < Spree::Base
     delegate :order, :currency, to: :shipment
     delegate :name, :tax_category, :tax_category_id, to: :shipping_method
     delegate :code, to: :shipping_method, prefix: true
-    alias_attribute :amount, :cost
 
-    alias_method :total_before_tax, :amount
+    alias_attribute :amount, :cost
+    def total_before_tax
```

Comment:
This is related to https://github.com/rails/rails/issues/49250. There does not seem to be a proposed solution, and apparently, this is the expected behavior. 

As you mentioned in the commit description, we could just use `cost`, but if anyone customized the `amount` method in their store, it's going to be breaking. Also, `amount` is used as `money_method` below, which we should also change, but the rest of Solidus uses `amount` as money method, so it's not ideal. 

I think that for now your solution is ok, thanks Peter.

Reply:
Looks like the proposed alternative in the rails ticket is to use `alias_attribute` twice, I'll update that